src/lib/webpack/webpack-base-config.js

```diff
@@ -97,6 +100,37 @@ export default (env) => {
 			}
 		}),
 
+		// TypeScript
+		customConfig({
+			module: {
+				loaders: [
+					{
+						enforce: 'pre',
+						test: /\.tsx?$/,
+						use: [
+							{
+								loader: resolve(__dirname, './npm-install-loader'),
+								options: {
+									modules: ['typescript', 'awesome-typescript-loader'],
+									save: true
+								}
+							},
+							{
+								loader: 'awesome-typescript-loader',
+								options: {
+									sourceMap: true,
+									useBabel: true,
+									babelOptions: createBabelConfig(env),
+									useCache: true,
+									configFileName: tsconfig,
```

Comment:
As a side-note, it's working fine for me with `ts-loader` via this `preact.config.js` hack:

```js
export default function (config, env, helpers) {
	config.module.loaders.push({
		enforce: "pre",
		test: /\.tsx?$/,
		loader: "ts-loader"
	});
}
```